kufar/books/main.go

```diff
@@ -59,14 +44,66 @@ type BookResponse struct {
 	Name   string `json:"name"`
 	Author string `json:"author"`
 	Date   string `json:"date"`
+	UserID int    `json:"userid"`
+}
+
+//CredentialsRequest ...
+type CredentialsRequest struct {
+	Username string `json:"username"`
+	Password string `json:"password"`
+}
+
+//Credentials ...
+type Credentials struct {
+	UUID     int
+	Username string
+	Password string
 }
 
 //Database ...
 type Database struct {
 	db *sql.DB
 }
 
-var location, _ = time.LoadLocation("Europe/Minsk")
+func main() {
+	r := chi.NewRouter()
+	db, err := New()
+	if err != nil {
+		log.Fatalf("error connecting to db: %v", err)
+	}
+
+	defer db.Close()
+
+	r.Post("/signup", SignUp(*db))
+	r.Post("/signin", SignIn(*db))
+	r.Post("/logout", Logout(*db))
+
+	r.Route("/books", func(r chi.Router) {
+		r.Post("/", Create(*db))
+		r.Get("/", ReadAll(*db))
+
+		r.Route("/{id}", func(r chi.Router) {
+			r.Get("/", Read(*db))
+			r.Put("/", Update(*db))
+			r.Delete("/", Delete(*db))
+		})
+	})
+
+	log.Fatal(http.ListenAndServe(":8080", r))
+}
+
+func getenv(key, fallback string) string {
+	if value, ok := os.LookupEnv(key); ok {
+		return value
+	}
+	return fallback
+}
+
+// returns true is the session is new and vice-versa
+func checkSession(w http.ResponseWriter, r *http.Request) (id interface{}, valid bool) {
+	session, err := store.Get(r, "my-cookie")
+	return session.Values["user"], err != nil || session.IsNew
```

Comment:
If session will be empty, you will get panic. You need to rewrite this return :)